lib/vanagon/component.rb

```diff
@@ -260,14 +260,28 @@ def fetch_mirrors(options)
           VanagonLogger.info %(Attempting to fetch from mirror URL "#{mirror}")
           @source = Vanagon::Component::Source.source(mirror, **options)
           return true if source.fetch
+        rescue Vanagon::Error => e
+          if e.message =~ /Unknown file type/
```

Comment:
I think I'd prefer raising an subclass of `Vanagon::Error` specifically for the unknown source case and rescue the most specific exceptions first like:

```ruby
rescue Vanagon::UnknownSourceError
  VanagonLogger.error %(Unable to access mirror URL "#{mirror}")
rescue Vanagon::Error 
  raise
rescue SocketError
  ...
rescue StandardError
  ...
```